haystack/components/routers/zero_shot_text_router.py

```diff
@@ -0,0 +1,177 @@
+from typing import Any, Dict, List, Optional
+
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.lazy_imports import LazyImport
+from haystack.utils import ComponentDevice, Secret, deserialize_secrets_inplace
+
+logger = logging.getLogger(__name__)
+
+
+with LazyImport(message="Run 'pip install transformers[torch,sentencepiece]'") as torch_and_transformers_import:
+    from transformers import pipeline
+
+    from haystack.utils.hf import (  # pylint: disable=ungrouped-imports
+        deserialize_hf_model_kwargs,
+        resolve_hf_pipeline_kwargs,
+        serialize_hf_model_kwargs,
+    )
+
+
+@component
+class TransformersZeroShotTextRouter:
+    """
+    Routes a text input onto different output connections depending on which label it has been categorized into.
+    This is useful for routing queries to different models in a pipeline depending on their categorization.
+    The set of labels to be used for categorization can be specified.
+
+    Example usage in a retrieval pipeline that passes question-like queries to an embedding retriever and keyword-like
+    queries to a BM25 retriever:
+
+    ```python
+    from haystack.core.pipeline import Pipeline
+    from haystack.components.routers import TransformersZeroShotTextRouter
+    from haystack.components.embedders import SentenceTransformersTextEmbedder
+
+    p = Pipeline()
+    p.add_component(instance=TransformersZeroShotTextRouter(labels=["passage", "query"]), name="text_router")
+    p.add_component(
+        instance=SentenceTransformersTextEmbedder(model="intfloat/e5-base-v2", prefix="passage: "),
+        name="passage_embedder"
+    )
+    p.add_component(
+        instance=SentenceTransformersTextEmbedder(model="intfloat/e5-base-v2", prefix="query: "),
+        name="query_embedder"
+    )
+    p.connect("text_router.passage", "passage_embedder.text")
+    p.connect("text_router.query", "query_embedder.text")
+    # Query Example
+    p.run({"text_router": {"text": "What's your query?"}})
+    # Passage Example
+    p.run({
+        "text_router":{
+            "text": "Last week I upgraded my iOS version and ever since then my phone has been overheating whenever I use your app."
+        }
+    })
+    ```
+    """
+
+    def __init__(
+        self,
+        labels: List[str],
+        multi_label: bool = False,
+        model: str = "MoritzLaurer/deberta-v3-base-zeroshot-v1.1-all-33",
+        device: Optional[ComponentDevice] = None,
+        token: Optional[Secret] = Secret.from_env_var("HF_API_TOKEN", strict=False),
+        pipeline_kwargs: Optional[Dict[str, Any]] = None,
```

Comment:
Can we rename this parameter to `huggingface_pipeline_kwargs`?
In the past, we have noticed that `pipeline_kwargs` can confuse users.

Reply:
Ahh I see is the confusion cause by the fact that users think it's kwargs for the haystack `Pipeline` object?